### ISSUE TITLE
[BUG] Error when use `dio.download`

### ISSUE THREAD

**Issue by nvbien2000** [devmuaz/awesome_dio_interceptor#7]
### Issue:

When use `download()` method of `Dio`, an exception occurs in `AwesomeDioInterceptor`.

### Reason:

`_logJson()` function, line 92. Something has gone wrong with `_jsonEncoder.convert`:

```
Unhandled exception: Converting object to an encodable object failed: Instance of 'ResponseBody'
#0 _JsonStringifier.writeObject (dart:convert/json.dart:793:7) 
#1 _JsonStringStringifier.printOn (dart:convert/json.dart:982:17)
#2 _JsonStringStringifier.stringify (dart:convert/json.dart:967:5) 
#3 JsonEncoder.convert (dart:convert/json.dart:345:30)
#4 AwesomeDioInterceptor.Eval () 
#5 AwesomeDioInterceptor._logJson (package:abcxyz/common/network/haha_interceptor.dart:92:25) 
#6 AwesomeDioInterceptor._logResponse (package:abcxyz/common/network/haha_interceptor.dart:191:5) 
#7 AwesomeDioInterceptor.onResponse (package:abcxyz/common/network/haha_interceptor.dart:235:5) 
#8 DioMixin.fetch.responseInterceptorWrapper.<anonymous closure>.<anonymous closure> (package:dio/src/dio_mixin.dart:409:17) 
#9 new Future.<anonymous closure> (dart:async/future.dart:256:37) 
#10 Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15) 
#11 _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19) 
#12 _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5) 
#13 _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)

```
### Example code:
```
Dio dio = Dio();
dio.interceptors.add(AwesomeDioInterceptor());
await dio.download(
      "https://pdfobject.com/pdf/sample.pdf",
      "haha.pdf",
)
```

**Comment by FlawLessx** [comment]
any updates on this?

**Comment by devmuaz** [comment]
Hi,

First, I apologize for the delay, I've been busy all the time for the past few months.

Happily, this issue has been fixed in the latest version `v1.3.0` where it no longer throws an exception when downloading a file. Instead, it displays the file type or the instance itself:

<img width="573" alt="Screenshot" src="https://github.com/user-attachments/assets/76213b7f-3fb7-4423-94bd-8cd5c2dcbc8a">

I'm gonna be closing this issue, but feel free to re-open it at any time,
Thank you!
